src/gpu/tsvd/tsvd.cu

```diff
@@ -133,6 +133,26 @@ namespace tsvd
 		} );
 	}
 
+	/**
```

Comment:
Is it better in perf or readability? Or both?

Reply:
Perf should be about the same. Transform is better for readability and also safer - no chance of an illegal memory access because of an indexing mistake.